### DISCUSSION TITLE
Collect and report statistics grouped by directory with Minitest

### DISCUSSION THREAD

**Discussion by lioneldebauge** [test-prof/test-prof#272] [Ideas]
Hello 👋 

First of all thanks for this awesome gem, its documentation and related articles. All those are really helpful when dealing with slow test suite.

It's a common pratice in Ruby/Rails world to group related tests in the same directory. In order to improve your test suite speed it is interesting to know what category of tests are the slowest (e.g. models, controller, etc.). 

For RSpec this is taken care of by TagProfiler taking advantage of the fact that to get support functionalities of rspec-rails a test need the appropriate metadata value with the key `type`.  For this same reason I assume a similar functionality is not available in Minitest.

Maybe we could generate for Minitest the same report as for TagProfiler but grouping simply by directory ? What do you think ?

I would be willing to open a PR if you think this feature suggestion makes sense

**Comment by palkan** [comment]
Hey!

First of all, thanks for the feedback.

> Maybe we could generate for Minitest the same report as for TagProfiler but grouping simply by directory ? What do you think ?

That's a great idea. We already have it in our roadmap: https://github.com/orgs/test-prof/projects/2 Not sure when 2.0 is coming, so, any help is appreciated.

**Comment by lioneldebauge** [comment]
Hey 👋 

Sorry for the delay but I have created changes in my local environment. Is there anything I need to do to push those on a remote branch and open a PR ?

I get this error
```
remote: Permission to test-prof/test-prof.git denied to lioneldebauge.
fatal: unable to access 'https://github.com/test-prof/test-prof.git/': The requested URL returned error: 403
```

Pulling from master works with no issues.

**Reply by palkan** [reply]
Hey!

You must fork the repo and push your branch to the fork; and then you'll be able to open a PR.

**Reply by lioneldebauge** [reply]
Thanks for the explanation. First open source contribution 😅 

Here is the PR: https://github.com/test-prof/test-prof/pull/283
